src/v0/destinations/google_adwords_offline_conversions/transform.js

```diff
@@ -59,7 +60,20 @@ const getConversions = (message, metadata, { Config }, event, conversionType) =>
     payload = constructPayload(message, trackCallConversionsMapping);
     endpoint = CALL_CONVERSION.replace(':customerId', filteredCustomerId);
   }
-
+  const consentObject = populateConsentForGoogleDestinations(message, conversionType);
+  if (Object.keys(consentObject)?.length > 0) {
+    if (payload?.conversions?.length > 0) {
+      if (conversionType === 'click' || conversionType === 'call') {
```

Comment:
If we can check for the supported conversion type ahead of this logic, we may not even need to compute the consent object itself.

Reply:
moved logic inside the respective util functions for creating the payloads

Reply:
done